hail/python/hailtop/batch/job.py

```diff
@@ -1077,6 +1077,16 @@ def run_async(*args, **kwargs):
             if isinstance(value, Job):
                 raise BatchException('arguments to a PythonJob cannot be other job objects.')
 
+        # Some builtins like `print` do not have signatures
+        try:
+            inspect.signature(unapplied).bind(*args, **kwargs)
+        except ValueError as e:
+            # Some builtins like `print` don't have a signature that inspect can read
+            if 'no signature found for builtin' not in e.args[0]:
+                raise e
+        except TypeError as e:
+            raise BatchException(f'Cannot call {unapplied.__name__} with the supplied arguments') from e
```

Comment:
Can we add what the expected signature was and what was passed in the error message? If not, no worries.

Reply:
The `TypeError` that I raise from here has that information, basically the same error that you'd get if you had just called the function. Here's an example:

```ipython
In [1]: import hailtop.batch as hb
b =
In [2]: b = hb.Batch()

In [3]: j = b.new_python_job()

In [4]: def foo(a, b):
   ...:     a + b
   ...:

In [5]: j.call(foo, 1)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
~/hail/hail/python/hailtop/batch/job.py in call(self, unapplied, *args, **kwargs)
   1081         try:
-> 1082             inspect.signature(unapplied).bind(*args, **kwargs)
   1083         except ValueError as e:

~/opt/miniconda3/envs/hail/lib/python3.7/inspect.py in bind(*args, **kwargs)
   3014         """
-> 3015         return args[0]._bind(args[1:], kwargs)
   3016

~/opt/miniconda3/envs/hail/lib/python3.7/inspect.py in _bind(self, args, kwargs, partial)
   2929                             msg = msg.format(arg=param.name)
-> 2930                             raise TypeError(msg) from None
   2931             else:

TypeError: missing a required argument: 'b'

The above exception was the direct cause of the following exception:

BatchException                            Traceback (most recent call last)
<ipython-input-5-63e0907b869d> in <module>
----> 1 j.call(foo, 1)

~/hail/hail/python/hailtop/batch/job.py in call(self, unapplied, *args, **kwargs)
   1086                 raise e
   1087         except TypeError as e:
-> 1088             raise BatchException(f'Cannot call {unapplied.__name__} with the supplied arguments') from e
   1089
   1090         def handle_arg(r):

BatchException: Cannot call foo with the supplied arguments
```

Reply:
Happy to put additional info in if that's not enough.